framework/Models/Dummy.py

```diff
@@ -60,6 +60,17 @@ def specializeValidateDict(cls):
     cls.validateDict['Input' ][0]['multiplicity'] = 1
     cls.validateDict['Output'][0]['type'        ] = ['PointSet','DataSet']
 
+  def copyModel(self, obj):
+    """
+      This method is aimed to copy the "obj" model in this instance
+      It is generally used for unpickling objects (models)
+      @ In, obj, instance, the instance of the object to copy from
+      @ Out, None
+    """
+    if obj.type != self.type:
+      self.raiseAnError(IOError,'Only objects of the same type can be copied! {} != {} !'.format(obj.type, self.type))
+    self.__dict__.update(obj.__dict__)
```

Comment:
This seems like a potentially painful avoidance of the setstate method. Maybe we should call a setstate method here that can be overloaded as needed?

Reply:
I am not sure what to do about this.  Since __setstate__ is private, it will not be inherited, so I think we might then need to implement it in every Model subclass if we use it instead of something like copyModel. (That said even if we leave it, ```copyModel``` should probably be ```_copyModel``` so people think twice about using it.)

Reply:
is `__setstate__` private? I think we're using inheritance in the SVL for get and set state currently, if I understand right.

Reply:
How about changing it to this:
```python
    try:
      #If __getstate__ and __setstate__ available, use them
      self.__setstate__(obj.__getstate__())
    except AttributeError:
      #Otherwise just use object's dictionary
      self.__dict__.update(obj.__dict__)
```
So for classes where ```__setstate__``` and ```__getstate__``` are defined, it will use them, but otherwise it can just use the dictionary.

Reply:
Updated docstring, renamed to _copyModel, and added setstate/getstate use.